packages/dashboard/src/components/internalDashboard/index.test.tsx

```diff
@@ -124,4 +129,86 @@ describe('InternalDashboard', () => {
     expect(container.querySelector('.viewport-selection')).toBeTruthy();
     expect(container.querySelector('.iot-dashboard-panes-area')).toBeFalsy();
   });
+
+  it('can toggle to readonly mode', function () {
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    const args = {
+      readOnly: false,
+      dashboardConfiguration: {
+        widgets: [],
+        viewport: { duration: '5m' },
+      },
+    };
+
+    act(() => {
+      ReactDOM.render(
+        <Provider store={configureDashboardStore(args)}>
+          <DndProvider
+            backend={TouchBackend}
+            options={{
+              enableMouseEvents: true,
+              enableKeyboardEvents: true,
+            }}
+          >
+            <InternalDashboard hasEditPermission={true} query={undefined} messageOverrides={DefaultDashboardMessages} />
+          </DndProvider>
+        </Provider>,
+        container
+      );
+    });
+
+    const actionsContainer = container.querySelector('.button-actions');
+    expect(actionsContainer).toBeTruthy();
+
+    act(() => {
+      if (!actionsContainer) throw new Error('No actions on dashboard');
+      const foundButton = wrapper(actionsContainer).findButton('[data-test-id="actions-toggle-read-only-btn"]');
+      foundButton?.click();
+    });
+
+    const dashboard = container.querySelector('.iot-dashboard');
+    expect(dashboard).toBeTruthy();
+  });
+
+  it('cannot toggle to edit mode if no edit permissions', function () {
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    const args = {
+      readOnly: true,
+      dashboardConfiguration: {
+        widgets: [],
+        viewport: { duration: '5m' },
+      },
+    };
+
+    act(() => {
+      ReactDOM.render(
+        <Provider store={configureDashboardStore(args)}>
+          <DndProvider
+            backend={TouchBackend}
+            options={{
+              enableMouseEvents: true,
+              enableKeyboardEvents: true,
+            }}
+          >
+            <InternalDashboard
+              hasEditPermission={false}
+              query={undefined}
+              messageOverrides={DefaultDashboardMessages}
+            />
+          </DndProvider>
+        </Provider>,
+        container
+      );
+    });
+
+    const actionsContainer = container.querySelector('.button-actions');
+    expect(actionsContainer).toBeFalsy();
+
+    const dashboard = container.querySelector('.iot-dashboard');
```

Comment:
I tihnk this class wraps both the readonly and edit versions of the dashboard. I was using `.iot-dashboard-panes-area` for this type of check since that component is only used in the edit version, it wraps the resizeable panes